Allow enchantment influencers to modify influenced enchantments

LogicTechCorp opened the pull request:
I would like to add compatibility to my [mod](https://github.com/LogicTechCorp/Reagenchant) for Matrix Enchanting, but there currently isn't a way for an enchantment influencer to define custom enchantments. This PR fixes that by allowing enchantment influencers to modify their influenced enchantments.

Vazkii commented:
Rough idea seems to be that you'd put special items to "unlock" enchantments? Seems fine but I'd prefer you put this in a separate interface (could be a subclass of the current one) so it doesn't break compatibility.

LogicTechCorp commented:
I made the requested change and added a way to modify an enchantment influencer's smoke particle color. I also fixed an issue where an incorrect block position variable was being used to get the enchant power bonus in the matrix enchanting table's animate tick.
